### ISSUE TITLE
Feature Request: Don't require semi-colons after bound-method declarations.

### ISSUE THREAD

**Issue by AndyMoreland** [palantir/tslint#1076]
#### TypeScript code being linted

```ts
private handleFoo = (newState: boolean) => {
    this.setState({ banana: newState });
}
```
#### Actual behavior

When I have the "semicolon" rule on, tslint complains of a missing semi-colon after the closing `}` in this method definition.

#### Expected behavior

I want a way to make the semi-colon after the closing `}` optional. I view this construct as a way of defining a /method/, not a /property/, and methods don't close with `;`.

**Comment by adidahiya** [comment]
We could change the `semicolon` rule to start being lenient here, or add a new `allow-member-functions` to enable leniency. My slight preference is for the former. Thoughts @JKillian?

**Comment by AndyMoreland** [comment]
I've pushed a PR that addresses this, I think. If the semi-colon rule is set to "enabled", it'll be lenient. When it's "always", it'll require them.
